xwiki-platform-core/xwiki-platform-tree/xwiki-platform-tree-webjar/src/main/webjar/tree.js

```diff
@@ -493,7 +493,12 @@ define(['jquery', 'JobRunner', 'jsTree', 'tree-finder'], function($, JobRunner)
       } else if (data.event && !$(data.event.target).hasClass('jstree-no-link') &&
           $(data.event.target).closest('.jstree-no-links').length === 0) {
         // The node selection was triggered by an user event and links are enabled.
-        window.location.href = selectedNode.a_attr.href;
+        // When pressing Ctrl key and clicking on a document tree entry that is a link, open link in new window / tab
```

Comment:
```suggestion
        // When pressing Ctrl key and clicking on a tree node that has a link, open the link in new window / tab.
```
This code is for the **generic** tree. There's no notion of document at this level. The document tree macro uses this code later on.

Reply:
Thank you Marius for the review, I updated this code comment accordingly.